server/webapp/WEB-INF/rails/webpack/config/loaders/type-declarations-for-css-modules.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs          = require("fs");
+const os          = require("os");
+const loaderUtils = require("loader-utils");
+const path        = require("path");
+
+// See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Reserved_keywords_as_of_ECMAScript_2015
+const RESERVED_WORDS = new Set([
+  "break",
+  "case",
+  "catch",
+  "class",
+  "const",
+  "continue",
+  "debugger",
+  "default",
+  "delete",
+  "do",
+  "else",
+  "export",
+  "extends",
+  "finally",
+  "for",
+  "function",
+  "if",
+  "import",
+  "in",
+  "instanceof",
+  "new",
+  "return",
+  "super",
+  "switch",
+  "this",
+  "throw",
+  "try",
+  "typeof",
+  "var",
+  "void",
+  "while",
+  "with",
+  "yield"
+]);
+
+const CSS_MODULE_KEY_RE  = /"([^\\"]+)":/g;
+const WORD_ONLY_CHARS_RE = /^\w+$/i;
+
+module.exports = function createTypeDeclarations(content, map, meta) {
+  this.cacheable();
+  const callback = this.async();
+
+  let typeDecls = "";
+  const options = loaderUtils.getOptions(this) || {};
+
+  if (options.banner && "string" === typeof options.banner) {
+    typeDecls += `${options.banner}\n`;
+  }
+
+  const cssClassnames = [];
+
+  let match;
+
+  // content includes a huge payload which introduces extra/unwanted keys in the *.scss..d.ts
+  // so we strip them out by focusing on the locals
+  const locals = content.split("// Exports\nexports.locals = ").pop() || "";
```

Comment:
I'm a bit worried about this rather overly specific regex.

Seems like [this implementation](https://github.com/seek-oss/css-modules-typescript-loader/blob/master/index.js#L70) splits on `exports.locals`, is there a reason why this is not good enough, or is there a worry that perhaps `exports.locals` might catch too much gunk that will need further filtering?

Reply:
@ketan well, I think `exports.locals` could match a css selector, especially being that the rendered CSS payload is included above that actual exports.

I could change this to be somewhat safer:

`content.split(/\nexports\.locals[\s]*=[\s]*\{/)`

That should be less brittle as it won’t rely on a comment and instead syntax that would be expected to stay the same over upgrades.

Reply:
Sounds good to me. Make the change and merge it?

Reply:
@ketan will do - thanks for looking!